src/transformers/cache_utils.py

```diff
@@ -386,20 +384,23 @@ def update(
         Return:
             A tuple containing the updated key and value states.
         """
-        new_cache_positions = cache_kwargs.get("position_ids")
+        new_cache_positions = cache_kwargs.get("cache_position")
         k_out = self.key_cache
         v_out = self.value_cache
 
         k_out[:, :, new_cache_positions] = key_states
         v_out[:, :, new_cache_positions] = value_states
 
-        self.seen_tokens += key_states.shape[-2]
+        self.seen_tokens += key_states.shape[2]
         return k_out, v_out
 
     def get_seq_length(self, layer_idx: Optional[int] = 0) -> int:
         """Returns the sequence length of the cached states that were seen by the model. `layer_idx` kept for BC"""
         return self.seen_tokens
 
+    def get_usable_length(self, new_sequence_length=None, layer_idx: Optional[int] = 0) -> int:
```

Comment:
the args `new_sequence_length` and `layer_idx` seems never used here, is that intended ?

Reply:
This is for compatibility with other cache classes that have this, thus when `get_usable_length` you don't want to error out